march_hardware/src/ethercat/ethercat_master.cpp

```diff
@@ -185,6 +189,16 @@ void EthercatMaster::ethercatLoop()
       total_loops = 0;
       not_achieved_count = 0;
     }
+
+    const auto delta_t = std::chrono::high_resolution_clock::now() - this->valid_slaves_timestamp_ms_;
+    const auto slave_lost_duration = std::chrono::duration_cast<std::chrono::milliseconds>(delta_t);
+    const std::chrono::milliseconds slave_watchdog_timeout(this->slave_watchdog_timeout_);
+
+    if (slave_lost_duration > slave_watchdog_timeout)
```

Comment:
I think this is temporarily fine, but I think it would be better if the error below contained what slaves were missing. (Ideally you could even keep track from which slave was missing for the most time instead of throwing the slave that was missing at the end of the watchdog timeout)

Reply:
Added the last slave which was not detected to the error message.